src/Microsoft.AspNet.Mvc.ModelBinding/Binders/FormCollectionModelBinder.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Core.Collections;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    /// <summary>
+    /// Modelbinder to bind form values to <see cref="IFormCollection"/>.
+    /// </summary>
+    public class FormCollectionModelBinder : IModelBinder
+    {
+        /// <inheritdoc />
+        public async Task<bool> BindModelAsync([NotNull] ModelBindingContext bindingContext)
+        {
+            if (bindingContext.ModelType != typeof(IFormCollection)
+                && bindingContext.ModelType != typeof(FormCollection))
+            {
+                return false;
+            }
+
+            var request = bindingContext.OperationBindingContext.HttpContext.Request;
+            if (request.HasFormContentType)
+            {
+                var form = await request.ReadFormAsync();
+                var formValuesLookup = form.ToDictionary(p => p.Key,
```

Comment:
I'd guess we're not going to run into the multiple keys eyes since ReadFormAsync returns a string[] array of values, but would be useful to cover this via a test.